crypto/crypto/src/multi_ed25519.rs

```diff
@@ -0,0 +1,547 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+//! This module provides an API for the accountable threshold multi-sig PureEdDSA signature scheme
+//! over the ed25519 twisted Edwards curve as defined in [RFC8032](https://tools.ietf.org/html/rfc8032).
+//!
+//! Signature verification also checks and rejects non-canonical signatures.
+
+use crate::ed25519::{
+    Ed25519PrivateKey, Ed25519PublicKey, Ed25519Signature, ED25519_PRIVATE_KEY_LENGTH,
+    ED25519_PUBLIC_KEY_LENGTH, ED25519_SIGNATURE_LENGTH,
+};
+use crate::{traits::*, HashValue};
+use anyhow::{anyhow, Result};
+use bit_vec::BitVec;
+use core::convert::TryFrom;
+use libra_crypto_derive::{DeserializeKey, SerializeKey, SilentDebug, SilentDisplay};
+use once_cell::sync::Lazy;
+use std::fmt;
+
+const MAX_NUM_OF_KEYS: usize = 32;
+const BITVEC_NUM_OF_BYTES: usize = MAX_NUM_OF_KEYS / 8;
+static BITVEC_FIRST_KEY_SET: Lazy<BitVec> =
+    Lazy::new(|| BitVec::from_bytes(&[0b1000_0000, 0b0000_0000, 0b0000_0000, 0b0000_0000]));
+
+/// Vector of private keys in the multi-key Ed25519 structure along with the threshold.
+#[derive(DeserializeKey, SilentDisplay, SilentDebug, SerializeKey)]
+pub struct MultiEd25519PrivateKey(Vec<Ed25519PrivateKey>, u8);
```

Comment:
Why not make this more explicit:
  struct {
    key_set,
    threshold, // what is threshold
}

Reply:
Good point. I originally used that, then changed my mind as I thought it's just two and obvious. Agreed that it's more clear to explicitly define param names, will do!